percona/controllers/replica/replica.go

```diff
@@ -0,0 +1,320 @@
+		for k, v := range cluster.Spec.PGReplicas.HotStandby.Labels {
+			labels[k] = v
+		}
+	}
+	annotations := make(map[string]string)
+	if cluster.Spec.PGReplicas.HotStandby.Annotations != nil {
+		for k, v := range cluster.Spec.PGReplicas.HotStandby.Annotations {
+			annotations[k] = v
+		}
+	}
+	storage := crv1.PgStorageSpec{}
+	if cluster.Spec.PGReplicas.HotStandby.VolumeSpec != nil {
+		storageName := getReplicaName(cluster, index)
+		if len(cluster.Spec.PGReplicas.HotStandby.VolumeSpec.Name) > 0 {
+			storageName = cluster.Spec.PGReplicas.HotStandby.VolumeSpec.Name
+		}
+		storage = crv1.PgStorageSpec{
+			Name:               storageName,
+			StorageClass:       cluster.Spec.PGReplicas.HotStandby.VolumeSpec.StorageClass,
+			AccessMode:         cluster.Spec.PGReplicas.HotStandby.VolumeSpec.AccessMode,
+			Size:               cluster.Spec.PGReplicas.HotStandby.VolumeSpec.Size,
+			StorageType:        cluster.Spec.PGReplicas.HotStandby.VolumeSpec.StorageType,
+			SupplementalGroups: cluster.Spec.PGReplicas.HotStandby.VolumeSpec.SupplementalGroups,
+			MatchLabels:        cluster.Spec.PGReplicas.HotStandby.VolumeSpec.MatchLabels,
+		}
+	}
+
+	replica.ObjectMeta.Name = labels[config.LABEL_NAME]
+	replica.ObjectMeta.Namespace = cluster.Namespace
+	if replica.ObjectMeta.Labels == nil {
+		replica.ObjectMeta.Labels = make(map[string]string)
+	}
+	for k, v := range labels {
+		replica.ObjectMeta.Labels[k] = v
+	}
+	if replica.ObjectMeta.Annotations == nil {
+		replica.ObjectMeta.Annotations = make(map[string]string)
+	}
+	for k, v := range annotations {
+		replica.ObjectMeta.Annotations[k] = v
+	}
+	if cluster.Spec.PGReplicas.HotStandby.Affinity != nil {
+		replica.Spec.NodeAffinity = cluster.Spec.PGReplicas.HotStandby.Affinity.NodeAffinity
+	}
+	replica.Spec.Name = labels[config.LABEL_NAME]
+	replica.Spec.ReplicaStorage = storage
+	replica.Spec.UserLabels = cluster.Spec.UserLabels
+	replica.Spec.ClusterName = cluster.Name
+	if len(cluster.Spec.PGReplicas.HotStandby.Expose.ServiceType) > 0 {
+		replica.Spec.ServiceType = cluster.Spec.PGReplicas.HotStandby.Expose.ServiceType
+	}
+
+	return replica
+}
+
+func createOrUpdateReplicaService(clientset kubeapi.Interface, cluster *crv1.PerconaPGCluster) error {
+	ctx := context.TODO()
+	service, err := getReplicaServiceObject(cluster)
+	if err != nil {
+		return errors.Wrap(err, "get replica service object")
+	}
+
+	oldSvc, err := clientset.CoreV1().Services(cluster.Namespace).Get(ctx, getReplicaServiceName(cluster.Name), metav1.GetOptions{})
+	if err != nil {
+		_, err = clientset.CoreV1().Services(cluster.Namespace).Create(ctx, &service, metav1.CreateOptions{})
+		if err != nil {
+			return errors.Wrap(err, "create replica service")
+		}
+		return nil
+	}
+
+	service.ResourceVersion = oldSvc.ResourceVersion
+	service.Spec.ClusterIP = oldSvc.Spec.ClusterIP
+	_, err = clientset.CoreV1().Services(cluster.Namespace).Update(ctx, &service, metav1.UpdateOptions{})
```

Comment:
any check if services are different?
it could cause problems when service is `LoadBalancer`